client/src/Views/ReportsPage/Charts/FillChart.vue

```diff
@@ -109,28 +136,36 @@ export default {
     series() {
       return this.formatSeries(this.data);
     },
-    initialStartDate() {
-      const startDate = new Date();
-      const pastDate = startDate.getDate() - this.dateRange;
-      startDate.setDate(pastDate);
-      return startDate.toDateString();
+    date() {
+      const { startDate, endDate } = this.$props.dateRange;
+      const [year, month] = endDate.split('-');
+      const numDaysInMonth = new Date(year, month, 0).getDate();
+      const start = moment(startDate + '-01');
+      const end = moment(endDate + '-' + numDaysInMonth);
+      const daysRange = end.diff(start, 'days');
+
+      return {
+        range: daysRange > 0 ? daysRange : 0,
+        startDate: parseInt(start.format('x')),
+        endDate: `${endDate}-${numDaysInMonth}`,
+      };
     },
   },
   methods: {
     formatSeries(data) {
-      // order chronologically and truncate data
-      const chronological = this.sortByTimeAndTruncate(data, this.dateRange);
+      // order chronologically
+      const chronological = data.sort((b, a) => a.date - b.date);
 
       // loop over chronological shoutouts,
       // fill object with company values for keys set to array of zeros length of date range
-      const series = this.createInitialSeries(chronological, this.dateRange);
+      const series = this.createInitialSeries(chronological, this.date.range);
 
-      // loop over with the dateRange, go through each day and check to see if any shoutouts match that day
+      // loop over with the date.range, go through each day and check to see if any shoutouts match that day
       // if yes, go into series object and incremement the corresponding element in correct value array
-      this.updateSeries(series, chronological, this.dateRange);
+      this.updateSeries(series, chronological, this.date.range);
 
       // loop over series object and create correctly formatted data series for graph
```

Comment:
I think so since it's pretty complicated. I also think that there may be room to refactor it to be more concise and not have to loop so many times, but until then I think the comments are helpful